### ISSUE TITLE
ERROR  Update failed: error decoding response body: operation timed out

### ISSUE THREAD

**Issue by LinXunFeng** [shorebirdtech/shorebird#1601]
**Description**

I added Shorebird to my hybrid app according to the [documentation](https://docs.shorebird.dev/guides/hybrid-app/ios) and created the release and patch, but I encountered an error after restarting the app.

Because the patch can work normally under the Android app, it should be able to rule out network problems.

**Expected Behavior**

Complete the download and installation of the patch normally.

**Screenshots**

<img width="1040" alt="image" src="https://github.com/shorebirdtech/shorebird/assets/19367531/7052a99c-f104-419e-a400-7cd272d77b9e">

**Additional Context**

Console

```shell
2024-01-03 18:37:55.838328+0800 xxx[623:70498] [VERBOSE0:shorebird.cc(151)] Shorebird updater: no active patch.
2024-01-03 18:37:55.838424+0800 xxx[623:70498] [VERBOSE0:shorebird.cc(155)] Starting Shorebird update
[00:00:00.002] (1701cb000) INFO   Sending patch check request: PatchCheckRequest { app_id: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", channel: "stable", release_version: "7.0.0+2", patch_number: None, platform: "ios", arch: "aarch64" }

[00:00:30.871] (1701cb000) ERROR  Update failed: error decoding response body: operation timed out
Caused by:
    operation timed out

[00:00:30.871] (1701cb000) INFO   Update thread finished with status: Update had error
```

Command

```shell
shorebird release ios-framework-alpha --release-version 7.0.0+2
shorebird patch ios-framework-alpha --release-version 7.0.0+2
```

**Comment by bryanoltman** [comment]
Some things to try to help pinpoint the issue:

1. Does your device have internet access?
2. Where are you in the world? We expect that Shorebird might not fully work in China, although if you're able to successfully publish a patch, I'm not sure this is the problem.
3. Can you share your app ID? This isn't secret and can help us diagnose the issue.

**Comment by LinXunFeng** [comment]
I turned off automatic update and can work normally through `shorebird_code_push` after clicking the specified button.
This should be caused by the network permissions unique to iPhone(China mainland).

Now I can install the patch normally, but I found that the entire iOS App is very laggy. In addition, the phone is extremely hot. It is normal to use the xcframework created by `flutter build ios-framework --no-profile --no-debug`.
